op-node/sources/l1_client.go

```diff
@@ -115,3 +119,27 @@ func (s *L1Client) L1BlockRefByHash(ctx context.Context, hash common.Hash) (eth.
 	s.l1BlockRefsCache.Add(ref.Hash, ref)
 	return ref, nil
 }
+
+// L1HotShotCommitmentsFromHeight returns an array of HotShot commitments to sequencer blocks
+// This is used in the derivation pipeline to validate sequencer batches in Espresso mode
+func (s *L1Client) L1HotShotCommitmentsFromHeight(firstBlockHeight uint64, numHeaders uint64, hotshotAddr common.Address) ([]espresso.NmtRoot, error) {
```

Comment:
Would it be better to have this take a `HotShot` instead of the address, so we don't have to create a new client every time?

Reply:
Actually I think its ok, we actually aren't creating a new client every time - `NewHotshot` just binds a variable to an existing l1 backend (ethclient)